src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy

```diff
@@ -15,22 +15,15 @@
  */
 package com.bmuschko.gradle.docker.tasks.container
 
-import com.bmuschko.gradle.docker.tasks.AbstractDockerRemoteApiTask
+import com.bmuschko.gradle.docker.tasks.image.DockerExistingImage
 import com.bmuschko.gradle.docker.utils.CollectionUtil
-import groovy.transform.CompileStatic
 import org.gradle.api.provider.ListProperty
 import org.gradle.api.provider.Property
-import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.Optional
 
-import java.util.concurrent.Callable
-
-class DockerCreateContainer extends AbstractDockerRemoteApiTask {
-    @Internal
-    final Property<String> imageId = project.objects.property(String)
-
+class DockerCreateContainer extends DockerExistingImage {
```

Comment:
Shouldn't it be:
```suggestion
class DockerCreateContainer extends DockerExistingContainer {
```

Reply:
No, you are creating a new container so cannot depend on an existing container ID. Here you need an image ID.